### ISSUE TITLE
Hyperparameter optimization suggests different hyperparameters for different metrics

### ISSUE THREAD

**Issue by borao** [chemprop/chemprop#89]
I am trying to optimize hyperparameters for my model but I get two different suggestions for each of the runs described below where the only difference is the metric. Given that the metric is not the loss function (should be using BCEWithLogitsLoss), why would I have such different results?

`chemprop_hyperopt --data_path ../data/IC50TrainingData.csv --dataset_type classification --num_iters 5 --config_save_path ic50h-roc2
`
```
{
    "depth": 3,
    "dropout": 0.15000000000000002,
    "ffn_num_layers": 1,
    "hidden_size": 1400
}
```

`chemprop_hyperopt --data_path ../data/IC50TrainingData.csv --dataset_type classification --num_iters 5 --metric prc-auc --config_save_path ic50h-prc2
`
```
{
    "depth": 4,
    "dropout": 0.2,
    "ffn_num_layers": 3,
    "hidden_size": 600
}
```

**Comment by swansonk14** [comment]
Hi @borao, thanks for bringing up the issue! I think the problem is that there isn't always a monotonic relationship between different metrics, meaning that if one metric is higher for a particular model, that doesn't always mean that the other metric will also be higher. I've actually often seen cases where model 1 might get a higher AUC than model 2, but model 2 gets a higher PRC-AUC that model 1. If that happens during hyperparameter optimization, then the hyperopt algorithm may end up finding a different set of optimal hyperparameters because a model with one set of hyperparameters may perform better or worse compared to a model with a different set of hyperparameters depending on the metric. Does that answer your question?

**Comment by borao** [comment]
Hi @swansonk14 yes, thank you. That makes more sense. I overlooked how in the hyperparameter optimization it would be using the metric to select the parameters... sorry, my brain was in regular training mode where I was thinking about it as just optimizing to the loss function. 

Thanks for all your work on chemprop- I've enjoyed using it!
